Sources/OPCore/networking/OPParser.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+
+import Endpoints
+
+public class OPParser<T: Decodable>: JSONParser<T> {
+    
+    public typealias OutputType = T
+    
+    override public var jsonDecoder: JSONDecoder {
+        return decoder
+    }
+    
+    private let decoder: JSONDecoder = {
+        let d = JSONDecoder()
+        
+        d.dateDecodingStrategy = .custom({ (decoder) -> Date in
+            let container = try decoder.singleValueContainer()
+            let dateStr = try container.decode(String.self)
+            
+            let formatters = [
+                "yyyy-MM-dd",
+                "yyyy-MM-dd'T'HH:mm:ssZZZZZ",
+                "yyyy-MM-dd'T'HH:mm:ss.SSSZZZZZ",
+                "yyyy-MM-dd'T'HH:mm:ss'Z'",
+                "yyyy-MM-dd'T'HH:mm:ss.SSS",
+                "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'",
+                "yyyy-MM-dd HH:mm:ss"
+            ].map { (format: String) -> DateFormatter in
+                let formatter = DateFormatter()
+                formatter.locale = Locale(identifier: "en_US_POSIX")
+                formatter.dateFormat = format
+                return formatter
+            }
+            
+            for formatter in formatters {
```

Comment:
```swift
guard let date = formatters.compactMap { $0.date(from: dateStr) }.first else { throw }
return date
```
would this also work? I mean it would iterate more often than your solution so I think yours is good